### PR TITLE
Breaking: no-confusing-arrow enable allowParens: true (fixes #11503)

### LINKED ISSUE #11503: no-confusing-arrow enable allowParens: true by default
**What rule do you want to change?**
no-confusing-arrow
**Does this change cause the rule to produce more or fewer warnings?**
fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
enable `allowParens: true` by default
**Please provide some example code that this change will affect:**

```js
var x = a => (1 ? 2 : 3);
```

**What does the rule currently do for this code?**
report an error.
**What will the rule do after it's changed?**
no error reported.
**Are you willing to submit a pull request to implement this change?**
yes.

### PR DESCRIPTION
**Pull request by aladdin-add**
…es#11503)

**What is the purpose of this pull request? (put an "X" next to item)**

[ x] Documentation update
[ x] Changes an existing rule ([template]

**What changes did you make? (Give an overview)**
fixes #11503 

**Is there anything you'd like reviewers to focus on?**
no.

### CODE REVIEW

**Comment by aladdin-add** [comment]
@platinumazure , good catch! I found the default value not applied, just pushed a commit to fix it.

File: tests/lib/rules/no-confusing-arrow.js [hunk 1]
```diff
@@ -23,45 +23,47 @@ ruleTester.run("no-confusing-arrow", rule, {
         "a => { return 1 ? 2 : 3; }",
         "var x = a => { return 1 ? 2 : 3; }",
         "var x = (a) => { return 1 ? 2 : 3; }",
-        { code: "var x = a => (1 ? 2 : 3)", options: [{ allowParens: true }] }
+        { code: "var x = a => (1 ? 2 : 3)", options: [{ allowParens: true }] },
+        "var x = a => (1 ? 2 : 3)"
```
Comment (mysticatea) [inline tests/lib/rules/no-confusing-arrow.js]: Would you add tests for `allowParens: false`?
Looks there are no tests for the option...

Comment (aladdin-add) [reply tests/lib/rules/no-confusing-arrow.js]: Actually there are. (in invalid tests)
I think there are no **valid tests** for the option, just as all the cases will report an error. 😄

Comment (mysticatea) [reply tests/lib/rules/no-confusing-arrow.js]: Originately, were the line 23-35 valid tests for `{allowParens:false}`?

Comment (aladdin-add) [reply tests/lib/rules/no-confusing-arrow.js]: oops, I missed it, will add it later.

**Pull request merged**